src/app/app-routing.module.ts

```diff
@@ -1,21 +1,23 @@
-import { NgModule } from '@angular/core';
-import { RouterModule, Routes } from '@angular/router';
-import {AppComponent} from "./app.component";
+import {NgModule} from '@angular/core';
+import {RouterModule, Routes} from '@angular/router';
 import {HomeContainerComponent} from "./hero/home/home-container/home-container.component";
 import {PopularContainerComponent} from "./hero/popular/popular-container/popular-container.component";
 import {PageNotFoundComponent} from "./user/page-not-found/page-not-found.component";
 import {WelcomeComponent} from "./user/welcome/welcome.component";
+import {DetailComponent} from "./shared/detail/detail.component";
 
 const routes: Routes = [
-  {path:"", redirectTo:"welcome",component:WelcomeComponent},
-  {path:"welcome",component:WelcomeComponent},
-  {path:"home",component:HomeContainerComponent},
-  {path:"popular",component:PopularContainerComponent},
-  {path: '**', component: PageNotFoundComponent }
+  {path: "welcome", component: WelcomeComponent},
+  {path: "home", component: HomeContainerComponent},
+  {path: "popular", component: PopularContainerComponent},
+  {path: "detail/:id", component: DetailComponent},
```

Comment:
Just another small change, can you create a detail container component? and that can have a route instead of the detail component